plugins/scanners/scanoss/src/main/kotlin/ScanOss.kt

```diff
@@ -99,4 +108,71 @@ class ScanOss(
         val endTime = Instant.now()
         return generateSummary(startTime, endTime, results)
     }
+
+    data class ProcessedRules(
+        val includeRules: List<Rule>,
+        val ignoreRules: List<Rule>,
+        val replaceRules: List<ReplaceRule>,
+        val removeRules: List<RemoveRule>
+    )
+
+    private fun buildSettingsFromORTContext(context: ScanContext): ScanossSettings {
+        val rules = processSnippetChoices(context.snippetChoices)
+        val bom = Bom.builder()
+            .ignore(rules.ignoreRules)
+            .include(rules.includeRules)
+            .replace(rules.replaceRules)
+            .remove(rules.removeRules)
+            .build()
+        return ScanossSettings.builder().bom(bom).build()
+    }
+
+    fun processSnippetChoices(snippetChoices: List<SnippetChoices>): ProcessedRules {
+        val includeRules = mutableListOf<Rule>()
+        val ignoreRules = mutableListOf<Rule>()
+        val replaceRules = mutableListOf<ReplaceRule>()
+        val removeRules = mutableListOf<RemoveRule>()
+
+        snippetChoices.forEach { snippetChoice ->
+            snippetChoice.choices.forEach { choice ->
+                when (choice.choice.reason) {
+                    SnippetChoiceReason.ORIGINAL_FINDING -> {
+                        includeRules.addFinding(choice)
```

Comment:
One more naming nit / question: When comparing this to line 144, it seems that maybe `addFinding()` should rather be called `includeFinding()`.

However, there's also a slight mismatch between "include" here and "remove" below. Usually, the wording pairs would be "add" / "remove" or "include" / "exclude". But you know better than me whether having strict opposite wording is even applicable here.

What do you think?

Reply:
I changed it to `includeFinding()` to match our "include/remove" pattern. 

It makes more sense since we're actually including findings as context rather than just adding them. 
Does that approach make sense to you?

Reply:
Yes, fine! Thanks for sharing some background info here.